### DISCUSSION TITLE
Why is yaml.stringify({time: "12:00"}) not adding quotes to the resulting value?

### DISCUSSION THREAD

**Discussion by aisamu** [eemeli/yaml#264] [Q&A] [answered]
Hi!

Ran into this today, and it's unclear to me what the interpreted data type is and how could I guide the library to add those!
Wouldn't the sexagesimal floating-point number format require more sections/colons? 

I tested it on https://www.json2yaml.com, and the quote were preserved. 

Apologies if that's a known YAML behaviour!

```js
❯ node
Welcome to Node.js v12.22.1.
> const yaml = require('yaml')

> console.log(yaml.stringify({time: "12:00"}))
time: 12:00

> console.log(yaml.stringify({time: "1200"}))
time: "1200"
```

Happy to provide more context/examples if that's helpful!

(Update: some context https://github.com/dependabot/dependabot-core/issues/3615)

**Comment by eemeli** [comment, marked as answer]
It's because `yaml` defaults to the "core" schema of YAML 1.2, which does not include sexagesimal numbers. If you're working with another YAML library that uses the YAML 1.1 schema, it might make good sense to use the `version: '1.1'` option:

```js
YAML.parse('time: 12:00')
> { time: '12:00' }
YAML.parse('time: 12:00', { version: '1.1' })
> { time: 720 }
YAML.parse('time: "12:00"', { version: '1.1' })
> { time: '12:00' }

YAML.stringify({ time: '12:00' })
> 'time: 12:00\n'
YAML.stringify({ time: '12:00' }, { version: '1.1' })
> 'time: "12:00"\n'
```

The logic here is that unless otherwise specified, if it looks like a string could be written as a plain scalar, we do that. But if re-parsing that output string would consider it as having a different type in the current schema, the value gets quoted. Hence the quotes around `'1200'`, as that would otherwise be considered a number using both schemas.

**Reply by aisamu** [reply]
That makes perfect sense, thank you!
